src/herder/TxSetFrame.cpp

```diff
@@ -832,10 +998,12 @@ TxSetFrame::isGeneralizedTxSet() const
 bool
 TxSetFrame::addTxsFromXdr(Application& app,
                           xdr::xvector<TransactionEnvelope> const& txs,
-                          bool useBaseFee, std::optional<int64_t> baseFee)
+                          bool useBaseFee, std::optional<int64_t> baseFee,
+                          Phase phase)
```

Comment:
I think this is a good place to verify phase consistency (soroban phase should have only soroban txs and vice versa). Maybe I've missed the check elsewhere though...

Reply:
Fixed.